File: eeducation/src/main/kotlin/rs/eeducation/model/Course.kt
package rs.eeducation.model

import java.util.*
import javax.persistence.*

@Entity
class Course(@Id
             @GeneratedValue
             var id: Long?,
             @OneToMany
             var lessons: Set<Lesson>,
             @ManyToMany(mappedBy = "courses")
             var students: Set<Student>,
             @ManyToMany(mappedBy = "courses")
             var teachers: Set<Teacher>,
             @ManyToOne
             var school: School?,
             @OneToMany
             var grades: Set<Grade>,
             @OneToMany
             var tests: Set<Test>,
             @OneToMany
             var absences: Set<Absence>,
             @Column
             var name: String,
             @Column
             var beginsDate: Date,
             @Column
             var endsDate: Date,
             @Column
             var isArchivated: Boolean,
             @Column
             var freeToJoin: Boolean,
             @OneToMany
             var studentsWaiting: Set<Student>,
             @Column
             var descr: String) {

}

File: eeducation-front/src/app/model/Course.ts
import {Grade} from './Grade';
import {Test} from './Test';
import {Absence} from './Absence';
import {Student} from './Student';
import {Lesson} from './Lesson';
import {School} from './School';
import {UserBasic} from './UserBasic';

export class Course {
  public id: number;
  public lessons: Array<Lesson>;
  public students: Array<Student>;
  public teachers: Array<UserBasic>;
  public school: School;
  public grades: Array<Grade>;
  public tests: Array<Test>;
  public absences: Array<Absence>;
  public name: string;
  public begins: Date;
  public ends: Date;
  public description: string;
  public studentsWaiting: Array<UserBasic>;

}
